lib/incoming_mail/handler.rb

```diff
@@ -63,14 +63,17 @@ def create_comment(msg)
       proposal = find_proposal(find_public_id(msg)) or return
       comment_text = find_comment_text(msg)
       comment_user = find_comment_user(msg)
+      comment = nil
       if proposal.has_subscriber?(comment_user)
         # already in the loop, just add comment.
-        proposal.comments.create(comment_text: comment_text, user: comment_user)
+        comment = proposal.comments.create(comment_text: comment_text, user: comment_user)
       else
         # yes, user adds self as observer, which also generates comment
         proposal.add_observer(comment_user, comment_user, comment_text)
-        proposal.comments.last
+        comment = proposal.comments.last
       end
+      Dispatcher.on_comment_created(comment) # sends email
```

Comment:
I would say changing this is not just an optimization; the way it is now is confusing because of the amount of logic in that one method. Not pointing the finger at you or anyone on this -- but since reading and understanding this method was part of reviewing this pull request, my feedback is that it is difficult to parse as-is. 

That being said, I hear you on the "let's get this in because it is important" front. I didn't realize that this was fixing something that is broken on production. Maybe we should create a github tag to indicate when there is urgency of that nature in a PR?

:ship:  code!